src/store/dialogsReducer.js

```diff
@@ -47,17 +47,24 @@ const initialState = {
 };
 
 const dialogsReducer = (state = initialState, action) => {
+  const stateCopy = {
+    ...state,
+    messages: [...state.messages],
+  };
+
   switch (action.type) {
-    case UPDATE_NEW_MESSAGE_BODY:
-      state.newMessageBody = action.body;
-      return state;
+    case UPDATE_NEW_MESSAGE_BODY: {
+      stateCopy.newMessageBody = action.body;
+      return stateCopy;
+    }
 
-    case SEND_MESSAGE:
+    case SEND_MESSAGE: {
       const body = state.newMessageBody;
 
-      state.newMessageBody = '';
-      state.messages.push({ id: 8, message: body });
-      return state;
+      stateCopy.newMessageBody = '';
+      stateCopy.messages.push({ id: 8, message: body });
```

Comment:
Consider getting rid of "id: 8" as it's a magic number.

Reply:
/fxd